consul/catalog_endpoint.go

```diff
@@ -159,16 +172,26 @@ func (c *Catalog) ServiceNodes(args *structs.ServiceSpecificRequest, reply *stru
 	}
 
 	// Get the nodes
-	state := c.srv.fsm.State()
-	err := c.srv.blockingRPC(&args.QueryOptions,
+	state := c.srv.fsm.StateNew()
+	err := c.srv.blockingRPCNew(
+		&args.QueryOptions,
 		&reply.QueryMeta,
-		state.QueryTables("ServiceNodes"),
+		state_store.NewMultiWatch(
```

Comment:
Not sure how you guys feel about this where we expose the tables here. I really didn't like how KV was special here, but I do see that this exposes lower-level details about the tables up. It would be easy to provide a string-based lookup thing like the old interface where you pass the name of the function you are calling here.